internal/query/projection/group.go

```diff
@@ -0,0 +1,170 @@
+			handler.NewColumn(GroupColumnSequence, handler.ColumnTypeInt64),
+			handler.NewColumn(GroupColumnState, handler.ColumnTypeEnum),
+		},
+			handler.NewPrimaryKey(GroupColumnInstanceID, GroupColumnID), // todo: review
+			handler.WithIndex(handler.NewIndex("resource_owner", []string{GroupColumnResourceOwner})),
+		),
+	)
+}
+
+func (g *groupProjection) Reducers() []handler.AggregateReducer {
+	return []handler.AggregateReducer{
+		{
+			Aggregate: group.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  group.GroupAddedEventType,
+					Reduce: g.reduceGroupAdded,
+				},
+				{
+					Event:  group.GroupChangedEventType,
+					Reduce: g.reduceGroupChanged,
+				},
+				{
+					Event:  group.GroupRemovedEventType,
+					Reduce: g.reduceGroupRemoved,
+				},
+			},
+		},
+		{
+			Aggregate: group.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  org.OrgRemovedEventType,
+					Reduce: g.reduceOwnerRemoved,
+				},
+			},
+		},
+		{
+			Aggregate: instance.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  instance.InstanceRemovedEventType,
+					Reduce: reduceInstanceRemovedHelper(GroupColumnInstanceID),
+				},
+			},
+		},
+	}
+}
+
+func (g *groupProjection) reduceGroupAdded(event eventstore.Event) (*handler.Statement, error) {
+	e, ok := event.(*group.GroupAddedEvent)
+	if !ok {
+		return nil, zerrors.ThrowInvalidArgumentf(nil, "PRJXN-rdz7Sy", "reduce.wrong.event.type %s", group.GroupAddedEventType)
+	}
+	return handler.NewCreateStatement(
+		e,
+		[]handler.Column{
+			handler.NewCol(GroupColumnID, e.Aggregate().ID),
+			handler.NewCol(GroupColumnName, e.Name),
+			handler.NewCol(GroupColumnResourceOwner, e.Aggregate().ResourceOwner),
+			handler.NewCol(GroupColumnInstanceID, e.Aggregate().InstanceID),
+			handler.NewCol(GroupColumnDescription, e.Description),
+			handler.NewCol(GroupColumnCreationDate, e.CreationDate()),
+			handler.NewCol(GroupColumnChangeDate, e.CreationDate()),
+			handler.NewCol(GroupColumnSequence, e.Sequence()),
+			handler.NewCol(GroupColumnState, domain.GroupStateActive),
+		},
+	), nil
+}
+
+func (g *groupProjection) reduceGroupChanged(event eventstore.Event) (*handler.Statement, error) {
+	e, ok := event.(*group.GroupChangedEvent)
+	if !ok {
+		return nil, zerrors.ThrowInvalidArgumentf(nil, "PRJXN-edc9Ay", "reduce.wrong.event.type %s", group.GroupChangedEventType)
+	}
+	if e.Name == "" {
+		return handler.NewNoOpStatement(e), nil
+	}
+
+	return handler.NewUpdateStatement(
+		e,
+		[]handler.Column{
+			handler.NewCol(GroupColumnName, e.Name),
+			handler.NewCol(GroupColumnDescription, e.Description),
+			handler.NewCol(GroupColumnChangeDate, e.CreationDate()),
+			handler.NewCol(GroupColumnSequence, e.Sequence()),
+		},
+		[]handler.Condition{
+			handler.NewCond(GroupColumnID, e.Aggregate().ID),
+			handler.NewCond(GroupColumnInstanceID, e.Aggregate().InstanceID),
+		},
+	), nil
```

Comment:
will need an update once my last comment on the previous PR is solved (name and description as pointers)